src/lmdb.ml

```diff
@@ -217,11 +216,15 @@ module Flags = struct
   let create     = mdb_CREATE
 end
 
+type buffer =
+  (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t
```

Comment:
This exactly corresponds to `Bigstringaf.t`, which exposes many intrinsic and optimized primitives for manipulating bigstrings. I think you could improve the rest of the code quite a lot by simply depending on that library.